src/coreclr/jit/compiler.cpp

```diff
@@ -2434,9 +2434,9 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
     {
         opts.compFlags = CLFLG_MINOPT;
     }
-    // Don't optimize .cctors (except prejit) or if we're an inlinee
+    // Don't optimize .cctors (except prejit and JitFullOptsForCctors) or if we're an inlinee
     else if (!jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT) && ((info.compFlags & FLG_CCTOR) == FLG_CCTOR) &&
```

Comment:
> This looks like an obsolete policy with tiering. Should cctors go through the regular tiereing process? Tier0, recompile with OSR if they happen to have a long running loop.

I agree, but we might want to disable PGO for it (Tier0 with loops are self-promoted to Tier0-instrumented) to avoid placing redundant probes (especially outside of loops).

What it might change is TieredCompilation=0 mode in which cctors will be compiled with FullOpts too, but, presumably, if user cares about startup time they won't disable `TieredCompilation`

Reply:
> What it might change is TieredCompilation=0 mode in which cctors will be compiled with FullOpts too, but, presumably, if user cares about startup time they won't disable TieredCompilation

Yes, I agree. @hez2010 What do you expect to see with this configuration switch? It does not look like particularly useful switch.

Reply:
We are doing minopts for all cctors today regardless of whether tiering is enabled. 

There're some GUI frameworks (such as winforms and WinUI 3) that are using cctors (or cctors generated by Roslyn for beforefieldinit) to do non-trivial things (like generating hundreds of delegates to build vtables for COMWrappers, initializing plenty of GUIDs/COM types, or using a loop and HW intrinsics to precompute a lookup table). In these cases minopts for cctors may instead slow down the startup due to low codegen quality.  

So I would like to try an option that enables fullopts for cctors only. Then we can profile and see how startup time will change if we are on a tiered enabled but fullopts for cctors only condition.

Reply:
> like generating hundreds of delegates to build vtables for COMWrappers,

This is straight-line code. Full opts are unlikely to make it cheaper overall. If the code runs once, full opts only help if they end up deleting large amounts of dead code, or if they 

> using a loop and HW intrinsics to precompute a lookup table

Is this loop actually source generated into the ctor? I would expect it to be in a helper function.

If you would into optimizing this, I think it would work better to analyze where the time goes in one of these static constructorsin detail. I doubt that this switch will provide you with useful insights. It will apply to all static constructors in the app and you won't be able to tell whether it actually helped somewhere.

Reply:
> I doubt that this switch will provide you with useful insights. It will apply to all static constructors in the app and you won't be able to tell whether it actually helped somewhere.

Sounds fair.
I changed my mind and instead just clean up the obsolete policy around cctors to allow them to be OSR-ed once there're long running loops.